### DISCUSSION TITLE
What is the best way to implement data separation?

### DISCUSSION THREAD

**Discussion by HokhyTann** [directus/directus#16984] [Archive]
Hello everyone!

I have been using and loving Directus so far. However, I've run into this design decision that I would like some ideas or suggestions from you guys or pls guide me to right discussion if existed.

So I am building this application with a set of user's roles and organization collection. All data and users should be belong to one organization exclusively. Hence, on each data collection, there is an organization_id field. I want to avoid cross-organization data access.

While, yes, I can separate db for each org. It is not cost-effective for me at this time. And I'm just thinking there might be a better way to do it.

Please give any ideas or suggestion! 
Thank everyone!

**Comment by licitdev** [comment]
Yes, you should be able to run your application within a single db.

The addition of the `organization_id` to each data collection is a great way to scope items down to a single organization.

By adding a `organization_id` to `directus_users`, you may use [Dynamic Variables](https://docs.directus.io/reference/filter-rules.html#dynamic-variables) within permissions.
Eg: `{ "organization_id": { "_eq": "$CURRENT_USER.organization_id" } }`

**Reply by HokhyTann** [reply]
Thank you so much for your confirmation. However, there are still two issues on Directus with this approach.

1. If a user is part of multiple organization, user-org is many-to-many, then, then we have no way to dynamically create the array of value for the custom filter "is one of". Both ui and api don't support such case.
2. When setting up the custom filter, the graphql marks the collection as unqueriable.

Do you have any idea about his @licitdev?

**Reply by licitdev** [reply]
1. You had mentioned that each user belongs exclusively to an organization. For multiple organizations, you may wish to refer to https://github.com/directus/directus/discussions/4959 instead.

2. Kindly provide more details for reproduction. You may wish to file a bug report instead. Thank you.

**Reply by HokhyTann** [reply]
1. Sorry, I contradicted myself a bit there. My use case is that all data except user should belongs exclusively to an org. User can belong to multiple organizations.
2. Yes, I'll try again and submit an issue if needed.

Thank you so much.
